devel/libffi/Portfile

```diff
@@ -67,4 +61,9 @@ depends_test-append port:expect port:dejagnu
 test.run            yes
 test.target         check
 
+# Some warnings need to be disabled when building tests with gcc-4.2
+if {[string match *gcc-4.2 ${configure.compiler}]} {
+    test.env-append LIBFFI_TEST_OLDGCC=1
```

Comment:
I thought about that, but decided to ignore it for two reasons:

1. There's no guarantee that these tweaks are adequate for gcc 4.0, so that would need to be tested.
2. MacPorts never builds this with gcc 4.0, anyway (that's what the `apple-gcc42` port is for).